be/src/exprs/aggregate_functions.h

```diff
@@ -372,6 +372,14 @@ class AggregateFunctions {
     static StringVal window_funnel_serialize(FunctionContext* ctx, const StringVal& src);
     static IntVal window_funnel_finalize(FunctionContext* ctx, const StringVal& src);
 
+    // retention
+    static void retention_init(FunctionContext* ctx, StringVal* dst);
+    static void retention_update(FunctionContext* ctx, int num_cond, const BooleanVal* conds,
+                                 StringVal* dst);
+    static void retention_merge(FunctionContext* ctx, const StringVal& src, StringVal* dst);
+    static StringVal retention_serialize(FunctionContext* ctx, const StringVal& src);
+    static BooleanVal* retention_finslize(FunctionContext* ctx, const StringVal& src);
+
     // todo(kks): keep following HLL methods only for backward compatibility, we should remove these methods
```

Comment:
We should add it in vectorized engine too.

Reply:
will do.

Reply:
The retention function vectorization has been completed, and the test code will be completed soon.

Reply:
`retention` function vectorization and all UT have been finished.